src/Entity/ItemId.php

```diff
@@ -95,16 +96,18 @@ public function unserialize( $serialized ) {
 	 * serialization. Not doing so in new code requires special justification.
 	 *
 	 * @param int|float|string $numericId
+	 * @param string $repositoryName, defaults to an empty string (local repository)
 	 *
 	 * @return self
 	 * @throws InvalidArgumentException
 	 */
-	public static function newFromNumber( $numericId ) {
+	public static function newFromNumber( $numericId, $repositoryName = '' ) {
 		if ( !is_numeric( $numericId ) ) {
 			throw new InvalidArgumentException( '$numericId must be numeric' );
 		}
+		RepositoryNameAssert::assertParameterIsValidRepositoryName( $repositoryName, '$repositoryName' );
```

Comment:
So I didn't change the constructor doc because ParameterAssertionException extends InvalidArgumentException. I believe using Assert's exception is OK as long as users expect InvalidArgumentException, no?